docs/DeveloperGuide.md

```diff
@@ -478,45 +483,105 @@ The redo does the exact opposite (pops from `redoStack`, push to `undoStack`, an
 - `UniqueTagList#editTagFromPerson(Person)` - Removes the original tags belonging to the specified person to the tag list and adds the new tags labelled for the specified person.
 - `UniqueTagList#getNumStudentsForTag(Tag)` - Returns the number of students labelled under the specified tag as stored in `tagCounter`.
-These operations are called when a person is added, deleted, or edited with `AddCommand`, `EditCommand` and `DeleteCommand`.
+- `UniqueTagList#asUnmodifiableTagList()` - Returns an unmodifiable view of the tag list.
+- `UniqueTagList#asUnmodifiableMap()` - Returns an unmodifiable view of the tag counter.
+These operations are called when a person is added, edited, or deleted with `AddCommand`, `EditCommand` and `DeleteCommand` respectively.
 
 Given below is an example usage scenario and how viewing tag is executed:
-- **Step 1:** The user launches the application. The `Model` is initialized with the saved data (or sample data if there is no saved data). Tags from each person is loaded into `UniqueTagList` and `tagCounter` with corresponding number of students labelled with the tags.
-- **Step 2:** The user enter the command `tag` to view all tags. `AddressBookParser` parses this command, creating a `TagCommand`.
-- **Step 3:** `LogicManager` executes the `TagCommand`. 
-  - During execution, `TagCommand#execute()` calls `Model#getObservableList()` to get the tag list with the tags created in TAB with their corresponding number of students labelled under them to be displayed to the user.
+- **Step 1:** The user launches the application. The `Model` is initialized with the saved data (or sample data if there were no saved data). Tags from each person is loaded into `UniqueTagList` and `tagCounter` with corresponding number of students labelled with the tags.
+- **Step 2:** The user enter the command `tag` to view all tags. `Logic` calls `AddressBookParser` to parse this command string, creating a `TagCommand`.
+- **Step 3:** `Logic` executes the `TagCommand`. During execution, `TagCommand#execute()` instantiates a `CommandResult` with the `DisplayType` of `TAGS` as a signal for `MainWindow` to switch the center panel to show the tag list.
+- **Step 4:** `MainWindow` then handles this command by calling `CenterPanel#displayTagListPanel()` to display the tag list to the user.
+
 <div markdown="span" class="alert alert-info">:information_source: **Note:** Tags with duplicate case-insensitive tag names for a person is not allowed. If the user tries to adds a tag with the same tag name to the person already with that tag, the new tag will not be added and `tagCounter` will not increment the count for this tag.<br></div>
 
 Figure I.4.1 shows a sequence diagram of how viewing tags works.<br>
-<img src="images/ViewTagSequenceDiagram.png" width="800" />
+<img src="images/ViewTagSequenceDiagramLogic.png" width="800" />
 
-*Figure I.4.1: View tag sequence diagram*
+*Figure I.4.1: View tags logic sequence diagram*
```

Comment:
Logic and CenterPanel shouldn't get deleted
![image](https://user-images.githubusercontent.com/70138507/140512678-9cb2f1a7-1dfe-4f16-92d2-7a87340374c0.png)